src/tests/nativeaot/SmokeTests/UnitTests/Generics.cs

```diff
@@ -3480,6 +3481,48 @@ public static void Run()
         }
     }
 
+    class Test99198Regression
+    {
+        delegate void Set<T>(ref T t);
+
+        interface IFoo
+        {
+            void Do<T>();
+        }
+
+        struct Foo<T> : IFoo
+        {
+            public nint Cookie1;
+            public nint Cookie2;
+
+            public void Do<T1>()
+            {
+                Cookie1 = 42;
+            }
+        }
+
+        class C<T> where T : IFoo
+        {
+            public static void Set(ref T t)
+            {
+                t.Do<T>();
```

Comment:
In general, looking at the fix, the only significant change I see is in `TypeBuilder.ParseNativeLayoutInfo` where we don't set the native function pointer on the non-template method in places where we used to do so, so I'd like to enhance this test to cover other cases where we might need to use the template.

Please update this test to make the Set method NoInline, as well as test the GVM pathway for invoking `Do<T>`.

``` csharp
[MethodImpl(MethodImplOptions.NoInlining)]
public static void Set(ref T t, IFoo ifoo)
{
    t.Do<T>();
    ifoo.Do<T>();
}
```

Then in the Run method, please allocate an instance of the Foo<object> type (or any other canonically compatible type) where we create the `EEType` for the IFoo instance dynamically instead of having the compiler generate it. (You're probably more familiar with the shenanigans to make that happen than I am at this point).

Please also test the case of calling the `Do<T>` method directly through reflection Invoke instead of going through the `Set` method.

Also, please take an audit to see what other paths use template methods, and see if they need to be tested here too.